### DISCUSSION TITLE
Token not available in health check request when using OAuth pass through

### DISCUSSION THREAD

**Discussion by nilsgaeq** [grafana/grafana#41235] [Data Sources]
Writing a backend datasource plugin, and enabling oauth pass through, the token of the grafana user is available in the `Authorization` header of the `QueryDataRequest`. However, the datasource also need to use this token while performing it's health check, verifying that everything is correctly set up towards the backend. This does not seem to be possible, as the token is not available in the `CheckHealthRequest`. Is there another way to get this token in the health check of the datasource?

**Comment by nilsgaeq** [comment]
The passhtru token is not available in `CallResourceRequest` either, it seems. Is it possible to get a reference to the `OAuthTokenService` in a plugin?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
